packages/asset-kit-block/src/components/AssetSelection.tsx

```diff
@@ -0,0 +1,77 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { BlockInjectButton } from '@frontify/guideline-blocks-shared';
+import { IconPlus24 } from '@frontify/fondue';
+import { useEffect, useState } from 'react';
+import { ASSET_SETTINGS_ID } from '../settings';
+import { AssetChooserObjectType, useAssetUpload, useFileInput } from '@frontify/app-bridge';
+import { AssetSelectionProps } from '../types';
+
+export const AssetSelection = ({
+    appBridge,
+    isUploadingAssets,
+    setIsUploadingAssets,
+    addAssetIdsToKey,
+    currentAssets,
+}: AssetSelectionProps) => {
+    const [openFileDialog, { selectedFiles }] = useFileInput({ multiple: true });
+    const [droppedFiles, setDroppedFiles] = useState<FileList | null>(null);
+    const [uploadFile, { results: uploadResults, doneAll }] = useAssetUpload({
+        onUploadProgress: () => !isUploadingAssets && setIsUploadingAssets(true),
+    });
+
+    const onOpenAssetChooser = () => {
+        appBridge.openAssetChooser(
+            (assetsObject) => {
+                setIsUploadingAssets(true);
+                const assetsIds = Array.from(assetsObject).map((asset) => asset.id);
+                addAssetIdsToKey(ASSET_SETTINGS_ID, assetsIds).then(() => setIsUploadingAssets(false));
+                appBridge.closeAssetChooser();
+            },
+            {
+                multiSelection: true,
+                selectedValueIds: currentAssets.map((asset) => asset.id),
+                objectTypes: [AssetChooserObjectType.ImageVideo],
+            }
+        );
+    };
+
+    useEffect(() => {
+        if (droppedFiles) {
+            setIsUploadingAssets(true);
+            uploadFile(droppedFiles);
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [droppedFiles]);
+
+    useEffect(() => {
+        if (selectedFiles) {
+            setIsUploadingAssets(true);
+            uploadFile(selectedFiles);
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [selectedFiles]);
+
+    useEffect(() => {
+        if (doneAll && uploadResults) {
+            (async (assetsObject) => {
+                const assetsIds = Array.from(assetsObject).map((asset) => asset.id);
+                addAssetIdsToKey(ASSET_SETTINGS_ID, assetsIds).then(() => setIsUploadingAssets(false));
```

Comment:
Minor:
use .finally instead of .then so the loading indication is stopped on error as well.

Reply:
good one, but i removed the `then(() =>` and used the await to match our code style